### PR TITLE
Link tracers

### PR DESCRIPTION
**Pull request by kshedstrom**
The linking that needed to happen was between tracers in the tracer registry and tracers in each segment tracer registry. Need to know the name of your tracers.

This is one step in the road to getting rid of the vardesc type.

### CODE REVIEW

**Comment by Hallberg-NOAA** [comment]
Thanks for this contribution, @kshedstrom. It looks like it will take a big step toward solving a number of our problems.

We might want to consider using doing a case-insensitive name comparison in tracer_name_lookup by doing lowercase() on both the name being queried and the one being sought.

**Comment by Hallberg-NOAA** [comment]
Pipeline testing underway at https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/3454

**Comment by kshedstrom** [comment]
So, it's a simple fix, but you already started the pipeline. I need to run my tests before pushing - would it bother things if I push before you do the merge?

**Comment by adcroft** [comment]
Adding commits to a PR is fine. We can restart the tests.

**Comment by kshedstrom** [comment]
Travis is timing out in the unit_tests. Not sure what to do about it.

**Comment by adcroft** [comment]
There's definitely something odd here. I've cleared cache and relaunched Travis for you but it is indeed hanging when it starts running. Latest of dev/gfdl worked fine but I don't see any differences that would do this...  :confused:

File: src/diagnostics/MOM_obsolete_params.F90 [hunk 1]
```diff
@@ -68,6 +69,12 @@ subroutine find_obsolete_params(param_file)
        hint="Instead use OBC_SEGMENT_XXX_DATA.")
   call obsolete_char(param_file, "EXTEND_OBC_SEGMENTS", &
        hint="This option is no longer needed, nor supported.")
+  call read_param(param_file, "OBC_NUMBER_OF_SEGMENTS", nseg)
```
Comment (adcroft) [inline src/diagnostics/MOM_obsolete_params.F90]: set `nseg = 0` before calling read_param() so that it is initialized for the loop that follows.

**Comment by kshedstrom** [comment]
Wow, thanks!

**Comment by adcroft** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/pipelines/3512

**Pull request merged**